src/image.rs

```diff
@@ -56,12 +58,21 @@ impl fmt::Display for ImageError {
     }
 }
 
-impl FromError<old_io::IoError> for ImageError {
-    fn from_error(err: old_io::IoError) -> ImageError {
+impl FromError<io::Error> for ImageError {
+    fn from_error(err: io::Error) -> ImageError {
         ImageError::IoError(err)
     }
 }
 
+impl FromError<byteorder::Error> for ImageError {
+    fn from_error(err: byteorder::Error) -> ImageError {
+        match err {
+            byteorder::Error::UnexpectedEOF => ImageError::ImageEnd,
```

Comment:
It seems it means the opposite. When it reads more data than allocated for the image. https://github.com/PistonDevelopers/image/blob/5df4d665ec2824b02e8ee0ac76c1d383a1e8e142/src/webp/decoder.rs#L112

@nwin Is this right?